src/librustdoc/clean/utils.rs

```diff
@@ -264,13 +263,13 @@ pub(crate) fn print_const(cx: &DocContext<'_>, n: ty::Const<'_>) -> String {
 }
 
 pub(crate) fn print_evaluated_const(tcx: TyCtxt<'_>, def_id: DefId) -> Option<String> {
-    tcx.const_eval_poly(def_id).ok().and_then(|val| {
+    tcx.const_eval_poly_for_typeck(def_id).ok().and_then(|val| {
```

Comment:
```suggestion
    tcx.const_eval_poly(def_id).ok().and_then(|val| {
```
These are not type system consts